src/main/java/com/pkb/unit/Unit.java

```diff
@@ -107,6 +107,84 @@ public Unit(String id, Bus bus, long retryPeriod, TimeUnit retryTimeUnit) {
         handleReportState();
     }
 
+
+    /**
+     * @return true if this set did not already contain the specified element
+     */
+    public void addDependency(String dependency) {
```

Comment:
Should we maybe implement this method in that way that it is in sync with Javadoc just to get some feedback? The more information, the better I think.

Reply:
I don't think the feedback is advantageous. If it's required we can add it later, for now it's just more to test I think. 
Good spot: I'll change the doc.